### ISSUE TITLE
Divide by zero error

### ISSUE THREAD

**Issue by tpederson-indeed** [ns1-terraform/terraform-provider-ns1#218]
Hi there,

### Terraform Version
```
Terraform v1.2.2
on darwin_amd64
+ provider registry.terraform.io/ns1-terraform/ns1 v1.12.7
```

### Affected Resource(s)
Please list the resources as a list, for example:
N/A

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### Debug Output
https://gist.github.com/tpederson-indeed/ddbc9563739e1b5824f85e03c1f98866

### Panic Output
No crash.log created

### Expected Behavior
The resources should have been created by the apply

### Actual Behavior
```
Stack trace from the terraform-provider-ns1_v1.12.7 plugin:

panic: runtime error: integer divide by zero

goroutine 84 [running]:
gopkg.in/ns1/ns1-go.v2/rest.RateLimit.WaitTime(...)
	gopkg.in/ns1/ns1-go.v2@v2.6.5/rest/client.go:310
gopkg.in/ns1/ns1-go.v2/rest.(*Client).RateLimitStrategyConcurrent.func1({0xc0007e2090, 0xc0004b2120, 0xc000310db0})
	gopkg.in/ns1/ns1-go.v2@v2.6.5/rest/client.go:334 +0x5f
gopkg.in/ns1/ns1-go.v2/rest.Client.Do({{0x1a34940, 0xc00010f260}, 0xc0003c6000, {0xc0004960d8, 0x14}, {0x18da620, 0xc}, 0xc000652130, 0x1, 0x0, ...}, ...)
	gopkg.in/ns1/ns1-go.v2@v2.6.5/rest/client.go:179 +0x124
gopkg.in/ns1/ns1-go.v2/rest.(*JobsService).Create(0xc0006c6248, 0xc0001c40e0)
	gopkg.in/ns1/ns1-go.v2@v2.6.5/rest/monitor_job.go:64 +0x1d3
github.com/terraform-providers/terraform-provider-ns1/ns1.MonitoringJobCreate(0x17f1620, {0x1846fa0, 0xc0006c6200})
	github.com/terraform-providers/terraform-provider-ns1/ns1/resource_monitoringjob.go:272 +0x6c
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00005ca00, 0xc00072a550, 0xc000932000, {0x1846fa0, 0xc0006c6200})
	github.com/hashicorp/terraform-plugin-sdk@v1.4.1/helper/schema/resource.go:305 +0x438
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00005d480, 0xc0003478d8, 0x18dc9bb, 0xf)
	github.com/hashicorp/terraform-plugin-sdk@v1.4.1/helper/schema/provider.go:294 +0x70
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00011c078, {0xc00080c1e0, 0x10af466}, 0xc00080c1e0)
	github.com/hashicorp/terraform-plugin-sdk@v1.4.1/internal/helper/plugin/grpc_provider.go:885 +0x7a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x18abc00, 0xc00011c078}, {0x1a48150, 0xc000494900}, 0xc00080c180, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.4.1/internal/tfplugin5/tfplugin5.pb.go:3189 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00037e2c0, {0x1a54488, 0xc0006c0000}, 0xc00066e400, 0xc000562cf0, 0x1f75120, 0x0)
	google.golang.org/grpc@v1.23.0/server.go:995 +0xe0d
google.golang.org/grpc.(*Server).handleStream(0xc00037e2c0, {0x1a54488, 0xc0006c0000}, 0xc00066e400, 0x0)
	google.golang.org/grpc@v1.23.0/server.go:1275 +0xa56
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.23.0/server.go:710 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.23.0/server.go:708 +0xef

Error: The terraform-provider-ns1_v1.12.7 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
 - Running on macos 12.3.1

### References
N/A

**Comment by tpederson-indeed** [comment]
I've been tearing apart my code bit by bit, and I've isolated it to this part.


filter_chains.tf:
```
resource "ns1_datafeed" "filter_chain" {
  for_each = var.monitor_to_ip_and_record

  name      = each.key
  source_id = ns1_datasource.datasource[var.record_to_pagerduty_service[each.value.record]].id

  config = {
    jobid = ns1_monitoringjob.filter_chain[each.key].id
  }
}

resource "ns1_monitoringjob" "filter_chain" {
  for_each = var.monitor_to_ip_and_record

  active          = true
  regions         = ["dal", "sin", "sjc", "lga", "ams"]
  job_type        = "tcp"
  frequency       = 60
  rapid_recheck   = true
  policy          = "quorum"
  notify_failback = true

  name = each.key

  # notify_list     = ns1_notifylist.notifylist[var.record_to_pagerduty_service[each.value.record]].id
  # notify_regional = false

  # rules {
  #   comparison = "contains"
  #   key        = "output"
  #   value      = "OK"
  # }
  # rules {
  #   comparison = "<"
  #   key        = "connect"
  #   value      = "5000"
  # }
  config = {
    ssl  = "1"
    # send = "GET /info/healthcheck?source=ns1/ HTTP/1.1\r\nHost: ${var.record_to_host_header[each.value.record]}\r\n\r\n"
    host = each.value.ip
    port = "443"
  }
}
```

variables.tf.json:
```
{
  "variable": {
    "monitor_to_ip_and_record": {
      "default": {
        "Americas Jobsearch pdxprod1-offload1001": {
          "ip": "35.165.48.217",
          "record": "americas"
        },
        "Americas Jobsearch pdxprod1-offload1002": {
          "ip": "35.162.238.247",
          "record": "americas"
        }
      }
    }
  }
}
```

I've been commenting out parts, but it seems to be throwing the divide by zero error from the main foreach:
```
resource "ns1_monitoringjob" "filter_chain" {
  for_each = var.monitor_to_ip_and_record
```

**Comment by tpederson-indeed** [comment]
I had it wrong above.  This works:
```
{
  "variable": {
    "monitor_to_ip_and_record": {
      "default": {
        "Americas Jobsearch pdxprod1-offload1001": {
          "ip": "35.165.48.217",
          "record": "americas"
        },
        "Americas Jobsearch pdxprod1-offload1002": {
          "ip": "35.162.238.247",
          "record": "americas"
        },
        "Americas Jobsearch pdxprod1-offload1003": {
          "ip": "54.203.219.73",
          "record": "americas"
        }
      }
    }
  }
}
```

This throws the error, however:
```
{
  "variable": {
        "Americas Jobsearch pdxprod1-offload1001": {
          "ip": "35.165.48.217",
          "record": "americas"
        },
        "Americas Jobsearch pdxprod1-offload1002": {
          "ip": "35.162.238.247",
          "record": "americas"
        },
        "Americas Jobsearch pdxprod1-offload1003": {
          "ip": "54.203.219.73",
          "record": "americas"
        },
        "Americas Jobsearch cmh-offload1001": {
          "ip": "18.216.114.70",
          "record": "americas"
        },
        "Americas Jobsearch cmh-offload1002": {
          "ip": "3.17.53.43",
          "record": "americas"
        },
        "Americas Jobsearch cmh-offload1003": {
          "ip": "18.218.22.89",
          "record": "americas"
        }
      }
    }
  }
}
```

**Comment by eravin-ns1** [comment]
Sorry for the delayed response. We haven't tracked this problem all the way to the bottom yet, but we've released v1.13.2-pre1 which should prevent the crash. As it's a prerelease, you'll need to specify the version explicitly in your resource file:
```
      source = "ns1-terraform/ns1"
      version = "1.13.2-pre1"
```
If this doesn't solve the problem, please generate logs with `TF_LOG=debug NS1_DEBUG=1 terraform apply ...` and submit them via an NS1 support ticket.

**Comment by eravin-ns1** [comment]
Release 1.13.2-pre2 should fix this by adding retries for 502/503 errors (which exposed the divide-by-zero bug fixed in 1.13.2-pre1).

**Comment by eravin-ns1** [comment]
Closing, this fixed in 1.13.2-pre2 and above.
